### DISCUSSION TITLE
How do I build the vscode-cpptools plugin from source?

### DISCUSSION THREAD

**Discussion by stravant** [microsoft/vscode-cpptools#9159] [Q&A]
When I clone the repo, open up `vscode-cpptools/Extention` in vscode, and hit run, it builds and launches the extension, but it doesn't work, I just get "Unable to start the C/C++ language server. IntelliSense features will be disabled. Error: Missing binary at d:\git\vscode-cpptools\Extension\bin\cpptools.exe". 

Clearly there's more to building the extension that I need to do but I can't find any documentation on how to build from source, did I miss some page explaining this?

**Comment by Colengms** [comment]
There is documentation in the repo, [here](https://github.com/microsoft/vscode-cpptools/blob/main/Documentation/Building%20the%20Extension.md).

**Reply by sean-mcmanus** [reply]
Also, it seems theoretically possible to add code to download the marketplace or GitHub vsix and automatically extract the binaries if they're not present, but I don't think it's currently worth the effort.
